### PR TITLE
RK-11578 - changes to transpilling docs

### PR DESCRIPTION
**Pull request by GuyGazit**

### CODE REVIEW

**Review by lirook — CHANGES_REQUESTED** [review]
I left some suggested changes. Let me know what you think

File: docs/node-setup.md [hunk 1]
```diff
@@ -93,25 +93,17 @@ If your application's code is being transpiled or bundled, you must include the
...
 - [**Babel**](https://babeljs.io/) - use either the `"inline"`, `"both"` or `true` values for the `sourceMaps` option in the Babel config file ([reference](https://babeljs.io/docs/en/options#sourcemaps)).
-- [**TypeScript**](https://www.typescriptlang.org/) - add either `"inlineSourceMap": true` or `"sourceMap": true` as well as `"inlineSources": true` in the TypeScript config file ([reference](https://www.typescriptlang.org/tsconfig#inlineSourceMap)).
+- [**TypeScript**](https://www.typescriptlang.org/) - add either `"inlineSourceMap": true` or `"sourceMap": true` in the TypeScript config file ([reference](https://www.typescriptlang.org/tsconfig#inlineSourceMap)).
 - [**CoffeeScript**](https://coffeescript.org/) - pass either the `-M` (inline) or `-m` flags to the `coffee` CLI tool ([reference](https://coffeescript.org/#usage)).
 
-### Advanced transpiling configurations
+### Don't Pack Rookout
 
-If multiple transpiling steps are used (for example - TypeScript followed by Webpack), you may need to add a step to your Webpack config. 
+Packing Rookout and other dependencies in general might create unnecessary issues, so we should leave them unpacked.
```
[the remark this answers is not in the document]
Comment (LiranLast) [reply docs/node-setup.md:101]: This statement is false :(

`and other dependencies in general might create unnecessary issues`

File: docs/node-setup.md [hunk 2]
```diff
@@ -93,25 +93,17 @@ If your application's code is being transpiled or bundled, you must include the
...
-2. Add the following rule to the webpack config file (under `rules`):
-
-```js
- {
-    test: /\.js$/,
-    use: ["source-map-loader"],
-    enforce: "pre"
- }
-```
+IMPORTANT: Don't forget to copy your `node_modules` directory to your server cause it is no longer packed.
```
[the remark this answers is not in the document]
Comment (LiranLast) [reply docs/node-setup.md:106]: This comment is not applicable to Serverless application which is where we are seeing the most use of Webpack...

File: docs/node-setup.md [hunk 3]
```diff
@@ -93,25 +93,17 @@ If your application's code is being transpiled or bundled, you must include the
 
 - [**Webpack**](https://webpack.js.org/) - use either the `inline-source-map` or `source-map` values for the `devtool` option in the webpack config file ([reference](https://webpack.js.org/configuration/devtool)).
 - [**Babel**](https://babeljs.io/) - use either the `"inline"`, `"both"` or `true` values for the `sourceMaps` option in the Babel config file ([reference](https://babeljs.io/docs/en/options#sourcemaps)).
-- [**TypeScript**](https://www.typescriptlang.org/) - add either `"inlineSourceMap": true` or `"sourceMap": true` as well as `"inlineSources": true` in the TypeScript config file ([reference](https://www.typescriptlang.org/tsconfig#inlineSourceMap)).
+- [**TypeScript**](https://www.typescriptlang.org/) - add either `"inlineSourceMap": true` or `"sourceMap": true` in the TypeScript config file ([reference](https://www.typescriptlang.org/tsconfig#inlineSourceMap)).
 - [**CoffeeScript**](https://coffeescript.org/) - pass either the `-M` (inline) or `-m` flags to the `coffee` CLI tool ([reference](https://coffeescript.org/#usage)).
 
-### Advanced transpiling configurations
+### Don't Pack Rookout
```
[the remark this answers is not in the document]
Comment (LiranLast) [reply docs/node-setup.md:99]: Let's make it softer?

We don't recommend it? Setting up Webpack to pack Rookout can get tricky?

File: docs/node-setup.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lirook) [reply docs/node-setup.md:101]: Which statement? @LiranLast Guy's, or my suggestion?

File: docs/node-setup.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lirook) [reply docs/node-setup.md:106]: @LiranLast it looks [here](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-package.html#nodejs-package-dependencies) like it is applicable, but I will make sure to also add documentation about our Lambda layer under the serverless section. WDYT?

File: docs/node-setup.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (LiranLast) [reply docs/node-setup.md:101]: That packaging deps is unnecessary :)

File: docs/node-setup.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lirook) [reply docs/node-setup.md:106]: Guy suggested to use `nodeExternals()` though, doesn't that mean we need everything from `node_modules`?

File: docs/node-setup.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lirook) [reply docs/node-setup.md:101]: Cool, so let's go with my copy :)

File: docs/node-setup.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (LiranLast) [reply docs/node-setup.md:106]: If that's why they choose to do.

We should probably also show how to exclude just Rookout (as Datadog does) as I believe at least some of customers will prefer not to do that.

[the remark this answers is not in the document]
Comment (lirook) [reply docs/node-setup.md:106]: I agree, testing locally I was able to do so with:
```javascript
  externals: {
    rookout: 'commonjs rookout'
  }
```

File: docs/node-setup.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (LiranLast) [reply docs/node-setup.md:101]: How about:

By default, Webpack will pack Rookout together with all other JS packages. Setting up packaging for the Rookout package can get tricky, and so we recommend excluding it by following these instructions:

**Review by lirook — APPROVED** [review]

**Pull request merged**